docs/_data/nakadi-event-bus-api.yaml

```diff
@@ -2053,6 +2053,13 @@ definitions:
           same partition and therefore compacted in a proper way (as compaction is performed per partition).
         type: string
         example: '329ed3d2-8366-11e8-adc0-fa7ae01bbebc'
+      span_ctx:
+        description: |
+          Object containing an OpenTracing http://opentracing.io span context. In case the producer of this event type
+          uses OpenTracing for tracing transcations accross distributed services, this field is be populated with an
+          object that allows consumers to resume the context of a span. The details of the payload are vendor
+          specific and consumers are expected to contact the producer in order to obtain further details.
+        type: object
```

Comment:
doesn't it have any defined structure?
(I ask this because we always criticize our users for not defining the schema properly :) )

Reply:
It is not possible or desired to define a schema as that would enforce vendor lock-in.

Unfortunately, [OpenTracing deliberately leaves the span context wire format undefined](https://github.com/opentracing/specification/issues/64). This allows each vendor to serialize/deserialize as they see fit. We want to keep the ability to switch vendors and should not depend on any particular format. 
Additionally, part of the span context are the [baggage items](https://github.com/opentracing/specification/blob/master/specification.md#set-a-baggage-item), which are arbitrary key values.
I remember this discussion from the initial narrative and a good idea could be to apply some sanity limits (payload size) for this particular element.

Reply:
https://github.com/zalando/nakadi/pull/931/files#r210564232

Reply:
I understand the reasons why it can't be defined.
I'm just afraid that it goes against what we fight for in Nakadi: all data to be defined. With this field it will never be possible to completely define the structure of the event.

Reply:
@v-stepanov I agree with you. Another point is that this will eventually find its way to the data lake, where incompatible changes in types might lead to breaking their processes. Maybe we should involve them here, so that they know about this new piece of information.

I would say that having arbitrary maps<string, string> is better than having arbitrary objects.

@lmineiro do you think it's possible to assume that the extensible parts fit in a map<string, string> or should we keep it as an opaque object, that might have arbitrary nested structures?

Reply:
Yes. We deliberately picked the [TextMap format](https://github.com/opentracing/specification/blob/master/specification.md#note-required-formats-for-injection-and-extraction) to, at least, provide that guarantee.

Reply:
If it is a `Map<String, String>`, it can be defined as such in JSON schema/OpenAPI schema:

```yaml
        type: object
        additionalProperties: string
```